flang/lib/Optimizer/Analysis/AliasAnalysis.cpp

```diff
@@ -27,6 +28,68 @@ using namespace mlir;
 
 #define DEBUG_TYPE "fir-alias-analysis"
 
+//===----------------------------------------------------------------------===//
+// AliasAnalysis: allocation detection based on MemAlloc effect
+//===----------------------------------------------------------------------===//
+
+static bool
+tryClassifyAllocateFromEffects(mlir::Operation *op, mlir::Value candidate,
```

Comment:
nit: isn't the candidate that may be accepted/rejected as an allocation the operation rather than the value?